WalletWasabi/Wallets/FileSystemBlockRepository.cs

```diff
@@ -19,11 +19,11 @@ public class FileSystemBlockRepository : IRepository<uint256, Block>
 	{
 		private const double MegaByte = 1024 * 1024;
 
-		public FileSystemBlockRepository(string blocksFolderPath, Network network, long targetBlocksFolderSizeMb = 300)
+		public FileSystemBlockRepository(string dataDir, Network network, long targetBlocksFolderSizeMb = 300)
 		{
 			using (BenchmarkLogger.Measure())
 			{
-				BlocksFolderPath = blocksFolderPath;
+				BlocksFolderPath = Path.Combine(dataDir, "BitcoinStore", network.ToString(), "Blocks");
```

Comment:
Why was hardcoding `BitcoinStore` here neccessary?

Reply:
dataDir identified as it is the only string in the container?

Reply:
We could have a DirectoryPathProvider class.

Reply:
> dataDir identified as it is the only string in the container?

Yes, it is only for demonstration purposes.

> We could have a DirectoryPathProvider class.

I would prefer to add some container for config values (`Config` class is not easily usable at this point). So a little bit more general concept than `DirectoryPathProvider`. But either way, it would be better with a class rather than with a string.

Reply:
> Why was hardcoding `BitcoinStore` here neccessary?

Yes, because now `string dataDir` is only in the container. This PR is a demo. It presents a concept rather than trying to be fully fledged solution.